contracts/cw20-streams/src/contract.rs

```
@@ -0,0 +1,575 @@
+
+    stream.claimed_amount += released;
+
+    STREAMS.save(deps.storage, id, &stream)?;
+
+    let config = CONFIG.load(deps.storage)?;
+    let cw20 = Cw20Contract(config.cw20_addr);
+    let msg = cw20.call(Cw20ExecuteMsg::Transfer {
+        recipient: stream.recipient.to_string(),
+        amount: released,
+    })?;
+
+    let res = Response::new()
+        .add_attribute("method", "withdraw")
+        .add_attribute("stream_id", id.to_string())
+        .add_attribute("amount", released)
+        .add_attribute("recipient", stream.recipient.to_string())
+        .add_message(msg);
+    Ok(res)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::GetConfig {} => to_binary(&query_config(deps)?),
+        QueryMsg::GetStream { id } => to_binary(&query_stream(deps, id)?),
+        QueryMsg::ListStreams { start, limit } => {
+            to_binary(&query_list_streams(deps, start, limit)?)
+        }
+    }
+}
+
+fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
+    let config = CONFIG.load(deps.storage)?;
+    Ok(ConfigResponse {
+        owner: config.owner.into_string(),
+        cw20_addr: config.cw20_addr.into_string(),
+    })
+}
+
+fn query_stream(deps: Deps, id: u64) -> StdResult<StreamResponse> {
+    let stream = STREAMS.load(deps.storage, id)?;
+    Ok(StreamResponse {
+        id,
+        owner: stream.owner.into_string(),
+        recipient: stream.recipient.into_string(),
+        amount: stream.amount,
+        claimed_amount: stream.claimed_amount,
+        rate_per_second: stream.rate_per_second,
+        start_time: stream.start_time,
+        end_time: stream.end_time,
+    })
+}
+
+fn query_list_streams(
+    deps: Deps,
+    start: Option<u8>,
+    limit: Option<u8>,
+) -> StdResult<ListStreamsResponse> {
+    let start_key = start.unwrap_or(0);
+    let start_bound = Bound::inclusive(start_key.to_string().as_bytes());
```

This will not work, we need bounds based on the actual key, not ascii form of it.

You can use [this list_proposals implementation](https://github.com/CosmWasm/cw-plus/blob/main/contracts/cw3-flex-multisig/src/contract.rs#L311-L326) as an example of an iterator over an integer key.

Also, collect into `StdResult<Vec<_>>` so we return an error if there is a parse error, not silently ignore it.